pages/create-bounty.tsx

```diff
@@ -0,0 +1,768 @@
+    setIsKyc(false);
+    setTierList([]);
+    setCurrentSection(0);
+    rewardERC20.setAddress(undefined);
+    transactionalERC20.setAddress(undefined);
+  }
+
+  useEffect(() => {
+    if(!connectedChain) return;
+
+    searchNetworks({
+      isRegistered: true,
+      chainId: connectedChain?.id,
+      sortBy: "name",
+      order: "asc",
+      isNeedCountsAndTokensLocked: true,
+    })
+      .then(async ({ count, rows }) => {
+        if (count > 0) {
+          setNetworks(rows);
+        }
+      })
+      .catch((error) => {
+        console.log("Failed to retrieve networks list", error);
+      });
+  }, [connectedChain]);
+
+  useEffect(() => {
+    if (transactionalToken?.address)
+      transactionalERC20.setAddress(transactionalToken.address);
+  }, [transactionalToken?.address, currentUser, Service?.active]);
+
+  useEffect(() => {
+    if (rewardToken?.address) rewardERC20.setAddress(rewardToken.address);
+  }, [rewardToken?.address, currentUser, Service?.active]);
+
+
+  useEffect(() => {
+    let approved = true;
+
+    if (!isFundingType)
+      approved = isAmountApproved(transactionalERC20.allowance,
+                                  BigNumber(issueAmount.value));
+    else if (rewardChecked)
+      approved = isAmountApproved(rewardERC20.allowance,
+                                  BigNumber(rewardAmount.value));
+
+    setIsTokenApproved(approved);
+  }, [
+    transactionalERC20.allowance,
+    rewardERC20.allowance,
+    issueAmount,
+    rewardAmount,
+    rewardChecked,
+  ]);
+
+  useEffect(() => {
+    if (currentNetwork && currentSection === 1 && connectedChain?.id) {
+      getReposList(true, currentNetwork.name, connectedChain?.id).then(setRepositories)
+    }
+  }, [currentNetwork, currentSection, connectedChain]);
+
+  useEffect(() => {
+    if (repository) {
+      getRepositoryBranches(repository.path, true).then((b) =>
+        setBranches(b.branches));
+    }
+  }, [repository]);
+
+  useEffect(() => {
+    if (!currentNetwork?.tokens) {
+      setTransactionalToken(undefined);
+      setRewardToken(undefined);
+      setCustomTokens([]);
+      transactionalERC20.setAddress(undefined);
+      rewardERC20.setAddress(undefined);
+      return;
+    }
+
+    if (!currentNetwork?.networkAddress || !connectedChain) return;
+
+    changeNetwork(connectedChain?.id, currentNetwork?.networkAddress);
+
+    const tokens = currentNetwork?.tokens
+
+    if (tokens.length === customTokens.length)
+      return;
+
+    if (tokens.length === 1) {
+      setTransactionalToken(tokens[0]);
+      setRewardToken(tokens[0]);
+    }
+
+    if (tokens.length === customTokens.length)
+      return;
+
+    setCustomTokens(currentNetwork?.tokens);
+  }, [currentNetwork, connectedChain]);
+
+  useEffect(() => {
+    cleanFields();
+    transactionalERC20.updateAllowanceAndBalance();
+    rewardERC20.updateAllowanceAndBalance();
+  }, []);
+
+  async function handleNetworkSelected(chain: SupportedChainData) {
```

Comment:
Like previously said, this function shouldn't be calling connect - instead it should instruct the connect to happen from the dropdown